sdk/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -116,7 +116,25 @@ public static Resource getTelemetrySdk() {
    */
   public static Resource create(Attributes attributes) {
     checkAttributes(Objects.requireNonNull(attributes, "attributes"));
-    return new AutoValue_Resource(attributes);
+    Attributes.Builder attrBuilder = Attributes.newBuilder();
+    attributes.forEach(new NullValueFilter(attrBuilder));
+    return new AutoValue_Resource(attrBuilder.build());
+  }
+
+  private static final class NullValueFilter implements KeyValueConsumer<AttributeValue> {
```

Comment:
I thought we got on the same page but see this didn't change yet - can't we filter in the producer here?

https://github.com/open-telemetry/opentelemetry-java/blob/master/api/src/main/java/io/opentelemetry/common/ImmutableKeyValuePairs.java#L58

Reply:
In the last SIG meeting, we agree on keeping the same behavior for all the interfaces that can create attributes.

Reply:
I may finally understand, is it that here

https://github.com/open-telemetry/opentelemetry-specification/blob/b4e327c3fbc62be811036cac572e4cc0d57dd560/specification/trace/api.md#set-attributes

where we discuss `Attribute values of null are considered to be not set and get discarded as if that SetAttribute call had never been made.`, this only refers to Spans? But Attributes are more general and could permit null values in other cases? It seems very confusing that we would support `null` attributes (not talking about as an implementation detail for modeling deletion for example but as actually exported value), but if that reading is correct than I agree we wouldn't filter in the producer, and may need to reclarify the spec.